homeassistant/components/tile/__init__.py

```diff
@@ -33,6 +33,31 @@ async def async_setup_entry(hass, entry):
     hass.data[DOMAIN][DATA_COORDINATOR][entry.entry_id] = {}
     hass.data[DOMAIN][DATA_TILE][entry.entry_id] = {}
 
+    # The existence of shared Tiles across multiple accounts requires an entity ID
+    # change:
+    #
+    # Old: tile_{uuid}
+    # New: {username}_{uuid}
+    #
+    # Find any entities with the old format and update them:
+    ent_reg = entity_registry.async_get(hass)
```

Comment:
We have this helper which may save some lines:
https://github.com/home-assistant/core/blob/1676bf220f187b808f874ff3eac654381bc5a147/homeassistant/helpers/entity_registry.py#L729

Reply:
Great to know, @MartinHjelmare! I'll follow up.